R/data.table.R

```diff
@@ -2681,14 +2681,21 @@ setnames = function(x,old,new,skip_absent=FALSE) {
   invisible(x)
 }
 
-setcolorder = function(x, neworder=key(x), before=NULL, after=NULL)  # before/after #4358
+setcolorder = function(x, neworder=key(x), before=NULL, after=NULL,skip_absent=FALSE)  # before/after #4358
 {
   if (is.character(neworder) && anyDuplicated(names(x)))
     stopf("x has some duplicated column name(s): %s. Please remove or rename the duplicate(s) and try again.", brackify(unique(names(x)[duplicated(names(x))])))
   if (!is.null(before) && !is.null(after))
     stopf("Provide either before= or after= but not both")
   if (length(before)>1L || length(after)>1L)
     stopf("before=/after= accept a single column name or number, not more than one")
+  if (!isTRUEorFALSE(skip_absent))
+    stopf("skip_absent should be TRUE or FALSE")
```

Comment:
This is a pretty common error, better for translation would be `stopf("%s should be TRUE or FALSE", "skip_absent")`, but that's out of scope for this PR.

Reply:
#6066

Reply:
#6067